addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py

```diff
@@ -458,14 +462,14 @@ def _invoice_constraints_peppol_en16931_ubl(self, invoice, vals):
                     'nl_r_004_street': self._check_required_fields(vals['customer'], 'street'),
                     'nl_r_004_city': self._check_required_fields(vals['customer'], 'city'),
                     'nl_r_004_zip': self._check_required_fields(vals['customer'], 'zip'),
-
+                })
+                error_msg = ""
+                for legal_entity in vals['vals']['accounting_customer_party_vals']['party_vals']['party_legal_entity_vals']:
                     # [NL-R-005] For suppliers in the Netherlands, if the customer is in the Netherlands,
                     # the customer’s legal entity identifier MUST be either a KVK or OIN number (schemeID 0106 or 0190)
-                    'nl_r_005': _(
-                        "The customer %s must have a KVK or OIN number.",
-                        vals['customer'].display_name
-                    ) if 'l10n_nl_oin' not in vals['customer']._fields or 'l10n_nl_kvk' not in vals['customer']._fields else '',
-                })
+                    if not legal_entity['company_id'] or legal_entity['company_id_attrs'] not in ['0106', '0190']:
+                        error_msg += _("The customer %s must have a KVK or OIN number. ", vals['vals']['accounting_customer_party_vals']['party_vals']['contact_vals']['name'])
+                constraints['nl_r_005'] = error_msg
```

Comment:
This is too complex, let's keep the old way and use the `vals['customer'].commercial_partner_id` instead of `vals['customer']` :)